src/models/sensor-gdx-manager.ts

```diff
@@ -0,0 +1,220 @@
+      let sensorConfig = new SensorConfiguration(this.internalConfig);
+      if (includeOnConnect) {
+        this.onSensorConnect(sensorConfig);
+      }
+      this.onSensorStatus(sensorConfig);
+    }
+
+    startPolling() {
+      setTimeout(() => {
+        this.sendSensorConfig(true);
+      }, 10);
+      setInterval(() => {
+        //TODO: do we need to cancel while collecting or while disconnected?
+        this.pollSensor();
+      }, 1000);
+
+    }
+
+    pollSensor() {
+      const readLiveData = async () => {
+        if (!this.disconnectRequested) {
+          this.enabledSensors.forEach((sensor: any, index: number) => {
+            const cNum = this.initialColumnNum + index;
+            this.internalConfig.columns[cNum].liveValueTimeStamp = new Date();
+            this.internalConfig.columns[cNum].liveValue = sensor.value.toString();
+          });
+          this.sendSensorConfig(false);
+        }
+      };
+      readLiveData();
+    }
+
+    hasSensorData() {
+      return this.hasData;
+    }
+
+    requestStart() {
+      console.log("Reading GDX measurements");
+
+      let startCollectionTime = Date.now();
+
+      const readData = async () => {
+        this.enabledSensors.forEach((sensor: any, index: number) => {
+          const cNum = this.initialColumnNum + index;
+          const time = Date.now() - startCollectionTime;
+          this.updateSensorValue(cNum.toString(), time / 1000, sensor.value);
+        });
+
+        if (!this.stopRequested) {
+          // Repeat
+          setTimeout(readData, 10);
+        } else {
+          this.onSensorCollectionStopped();
+          this.stopRequested = false;
+        }
+      };
+
+      readData();
+
+    }
+
+    updateSensorValue(ID:string, time:number, value:number) {
+      if (!value) {
+        return;
+      }
+      this.internalConfig.columns[ID].liveValue = value.toString();
+      this.hasData = true;
+
+      this.onSensorStatus(new SensorConfiguration(this.internalConfig));
+      const data:NewSensorData = {};
+      data[ID] = [[time, value]];
+      this.onSensorData(data);
+    }
+
+    requestStop() {
+      this.stopRequested = true;
+    }
+
+    async getBatteryLevel() {
+      const batteryLevel = await this.gdxDevice.getBatteryLevel();
+      console.log("Battery Level: " + batteryLevel);
+      return batteryLevel;
+    }
+
+    async connectToDevice(device?: any): Promise<boolean> {
+      this.gdxDevice = await godirect.createDevice(device);
+
+      if (!this.gdxDevice) {
+        console.log("Could not create GDX device");
+        return false;
+      }
+
+      console.log("Created and connected to GDX device " + this.gdxDevice.name);
+      console.log(this.gdxDevice);
+
+      // log disconnection
+      this.gdxDevice.on("device-closed", () => {
+        console.log("Disconnected from GDX device " + this.gdxDevice.name);
+      });
+
+      // turn on any default sensors
+      this.gdxDevice.enableDefaultSensors();
```

Comment:
Maybe not needed with the code below, but I'm following the lead from the Vernier example.